actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1584,6 +1584,14 @@ def shallow?
           !parent_resource.singleton? && @scope[:shallow]
         end
 
+        # Loads another route file located inside the +config/routes+ directory
+        # or any subdirectory, that has the same name as the argument given
+        # (+admin.rb+ in the example).
+        #
+        # In that file, you can use the normal routing DSL, but *do not* surround it
+        # with the +Rails.application.routes.draw+ block.
+        #
+        #   draw :admin
```

Comment:
I see this is mostly from https://guides.rubyonrails.org/v7.0/routing.html#breaking-up-very-large-route-file-into-multiple-small-ones, which is good.  However, the way the example is written here loses some of the context from the guide.  It looks as if "*do not* surround it with the `Rails.application.routes.draw` block" is referring to `draw :admin`.  I think we should expand the example to clarify.

Also, I would like to clarify that draw doesn't search every `config/routes` subdirectory.  Rather, it looks for `config/routes/#{name}.rb`, and `name` may include a subdirectory.

Lastly, I think it would be good to include the admonition from https://guides.rubyonrails.org/v7.0/routing.html#don-t-use-this-feature-unless-you-really-need-it.

So, altogether, I'm thinking something like:

```suggestion
        # Loads another routes file with the given +name+ located inside the
        # +config/routes+ directory. In that file, you can use the normal
        # routing DSL, but <i>do not</i> surround it with a
        # +Rails.application.routes.draw+ block.
        #
        #   # config/routes.rb
        #   Rails.application.routes.draw do
        #     draw :admin                 # Loads `config/routes/admin.rb`
        #     draw "third_party/some_gem" # Loads `config/routes/third_party/some_gem.rb`
        #   end
        #
        #   # config/routes/admin.rb
        #   namespace :admin do
        #     resources :accounts
        #   end
        #
        #   # config/routes/third_party/some_gem.rb
        #   mount SomeGem::Engine, at: "/some_gem"
        #
        # <b>CAUTION:</b> Use this feature with care. Having multiple routes
        # files can negatively impact discoverability and readability. For most
        # applications — even those with a few hundred routes — it's easier for
        # developers to have a single routes file.
```

What do you think?

Reply:
Yes! Sounds good, thank you for reviewing :)